Provide a clean compile time message if platform is neither x86/SSE2 nor ARM

You include SIMDE [here](https://github.com/surge-synthesizer/surge/blob/main/src/common/globals.h#L43) for x86 on riscv64 architecture.

Please check architecture in the cmake script during the configure phase.

error:
```
In file included from /wrkdirs/usr/ports/audio/surge-synthesizer-lv2/work/surge-release_1.9.0/src/common/dsp/effect/BBDEnsembleEffect.cpp:16:
In file included from /wrkdirs/usr/ports/audio/surge-synthesizer-lv2/work/surge-release_1.9.0/src/common/dsp/effect/BBDEnsembleEffect.h:22:
In file included from /wrkdirs/usr/ports/audio/surge-synthesizer-lv2/work/surge-release_1.9.0/src/common/dsp/VectorizedSvfFilter.h:3:
In file included from /wrkdirs/usr/ports/audio/surge-synthesizer-lv2/work/surge-release_1.9.0/src/common/vt_dsp/portable_intrinsics.h:4:
In file included from /nxb-bin/usr/lib/clang/14.0.5/include/immintrin.h:21:
/nxb-bin/usr/lib/clang/14.0.5/include/mmintrin.h:71:12: error: use of undeclared identifier '__builtin_ia32_vec_ext_v2si'
    return __builtin_ia32_vec_ext_v2si((__v2si)__m, 0);
           ^
/nxb-bin/usr/lib/clang/14.0.5/include/mmintrin.h:101:1: error: functions that differ only in their return type cannot be overloaded
_mm_cvtm64_si64(__m64 __m)
^
/usr/local/include/simde/x86/mmx.h:681:30: note: expanded from macro '_mm_cvtm64_si64'
#  define _mm_cvtm64_si64(a) simde_mm_cvtm64_si64(a)
                             ^
/usr/local/include/simde/x86/mmx.h:661:1: note: previous definition is here
simde_mm_cvtm64_si64 (simde__m64 a) {
^
```

--- a/src/common/globals.h
+++ b/src/common/globals.h
@@ -43,7 +43,7 @@
 #define SIMDE_ENABLE_NATIVE_ALIASES
 #include "simde/x86/sse2.h"
 #else
-#error SURGE requires either X86/SSE2 or ARM architectures.
+#error Surge XT requires either X86/SSE2 or ARM architectures.
 #endif
 #endif
 

--- a/src/common/globals.h
+++ b/src/common/globals.h
@@ -39,8 +39,12 @@
     (defined(_M_IX86_FP) && _M_IX86_FP >= 2)
 #include <emmintrin.h>
 #else
+#if defined(__arm__) || defined(__aarch64__)
 #define SIMDE_ENABLE_NATIVE_ALIASES
 #include "simde/x86/sse2.h"
+#else
+#error SURGE requires either X86/SSE2 or ARM architectures.
+#endif
 #endif
 
 #if MAC || LINUX

